jetty-io/src/main/java/org/eclipse/jetty/io/DefaultRetainableByteBufferPool.java

```diff
@@ -0,0 +1,314 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.io;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+
+import org.eclipse.jetty.util.BufferUtil;
+import org.eclipse.jetty.util.Pool;
+import org.eclipse.jetty.util.annotation.ManagedAttribute;
+import org.eclipse.jetty.util.annotation.ManagedObject;
+import org.eclipse.jetty.util.annotation.ManagedOperation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ManagedObject
+public class DefaultRetainableByteBufferPool implements RetainableByteBufferPool
```

Comment:
previously we have named our pools for their implementation and not for their usage.  This really is an `ArrayRetainableByteBufferPool` and if in future we want to switch to a Mapped pool, then the name doesn't make sense.

```suggestion
public class ArrayRetainableByteBufferPool implements RetainableByteBufferPool
```

Reply:
Ok, I'm happy with the new name.